src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/restrictions/job/StartedByMemberOfGroupRestriction.java

```diff
@@ -90,15 +102,39 @@ protected boolean acceptsUser(@CheckForNull String userId) {
         return false;
     }
 
-    
+    private List<String> getAuthorities(String userId) {
+        final @CheckForNull User usr = User.get(userId, false, null);
+        if (usr == null) {
+            return getAuthoritiesFromRealm(userId);
+        }
+        List<String> authorities = usr.getAuthorities();
+        if (authorities == null || authorities.size() == 0) {
```

Comment:
I don't know about the list if it can be null, if it can't of course I can remove the check.
The only way I can think of to manage valid empty lists is to introduce a checkbox to not check realm in the config - do you have any ideas? The other approach would be to take the draft code I have in my main that fails the job with a permission message in the log file instead of putting it back in the queue(then this would only be called once per job). But that feels like(like we discussed before a separate jira) :-/. I have some memory back in my mind that some realms (like the ldap realm or AD realm) also cache data but I'm unsure.